frontend/src/app/models/state.ts

```diff
@@ -0,0 +1,16 @@
+import { Page } from "./page";
+
+export interface State<T>{
+  data: Page<T>;
```

Comment:
You don't need to save "data" in state, because data is dynamic and frequently updated
Also see no reasons to save "loading" and "error" in state